src/software/firmware/srcs/alarm_controller.cpp

```diff
@@ -0,0 +1,189 @@
+
+          /**
+           * RCM-SW-19
+           * The device shall embed a low priority alarm 24 when the Plateau pressure < 2cm H2O at
+           * the second cycle (Patient disconnection).
+           */
+          Alarm(AlarmPriority::ALARM_MEDIUM, 24u, 2u),
+
+          /**
+           * RCM-SW-16
+           * The device shall embed an information (audible) signal 31 when the mains are
+           * disconnected to alert the user.
+           */
+          Alarm(AlarmPriority::ALARM_LOW, 31u, 1u)}) {}
+
+void AlarmController::snooze() {
+    // TODO disable buzzer sound for 120s
+}
+
+void AlarmController::detectedAlarm(uint8_t p_alarmCode, uint32_t p_cycleNumber) {
+    for (int i = 0; i < ALARMS_SIZE; i++) {
+        Alarm* current = &m_alarms[i];
+        if (current->getCode() == p_alarmCode) {
+            current->detected(p_cycleNumber);
+            break;
+        }
+    }
+}
+
+void AlarmController::notDetectedAlarm(uint8_t p_alarmCode) {
+    for (int i = 0; i < ALARMS_SIZE; i++) {
+        Alarm* current = &m_alarms[i];
+        if (current->getCode() == p_alarmCode) {
+            current->notDetected();
+            break;
+        }
+    }
+}
+
+void AlarmController::manageAlarm(uint16_t p_centiSec) {
+    AlarmPriority highestPriority = AlarmPriority::ALARM_NONE;
+    uint8_t triggeredAlarmCodes[ALARMS_SIZE];
+    uint8_t numberOfTriggeredAlarms = 0;
+
+    for (int i = 0; i < ALARMS_SIZE; i++) {
+        Alarm* current = &m_alarms[i];
+        if (current->isTriggered()) {
+            if (numberOfTriggeredAlarms == 0) {
+                highestPriority = current->getPriority();
+            }
+
+            triggeredAlarmCodes[numberOfTriggeredAlarms] = current->getCode();
+            numberOfTriggeredAlarms++;
+        }
+    }
+    displayAlarmInformation(triggeredAlarmCodes, numberOfTriggeredAlarms);
+
+    if (!highestPriority) {
+        Buzzer_Stop();
+
+        digitalWrite(PIN_LED_RED, LOW);
+        digitalWrite(PIN_LED_YELLOW, LOW);
+    } else if (highestPriority == AlarmPriority::ALARM_HIGH) {
+        if (m_highestPriority != highestPriority) {
+            Buzzer_Long_Start();
+        }
+
+        digitalWrite(PIN_LED_RED, HIGH);
+        digitalWrite(PIN_LED_YELLOW, LOW);
+    } else if (highestPriority == AlarmPriority::ALARM_MEDIUM) {
+        if (m_highestPriority != highestPriority) {
+            Buzzer_Medium_Start();
+        }
+        digitalWrite(PIN_LED_RED, LOW);
+        if (p_centiSec % 100 == 50) {
+            digitalWrite(PIN_LED_YELLOW, HIGH);
+        } else if (p_centiSec % 100 == 0) {
+            digitalWrite(PIN_LED_YELLOW, LOW);
+        }
+    } else if (highestPriority == AlarmPriority::ALARM_LOW) {
+        if (m_highestPriority != highestPriority) {
+            Buzzer_Short_Start();
+        }
+
+        digitalWrite(PIN_LED_RED, LOW);
+        digitalWrite(PIN_LED_YELLOW, HIGH);
+    }
+
+    m_highestPriority = highestPriority;
+
+    // TODO launch corresponding buzzer
```

Comment:
I was wondering: if the alarm last only one cycle for example, the buzzer will ring for a short time and on next cycle the alarm will be silenced and not displayed anymore. Shouldn't we keep the last alarm displayed?

Reply:
IIUC, this is not a problem. 
cc @cmdm can you confirm?

Reply:
I confirm: this is not a problem.